.github/workflows/build.yml

```diff
@@ -44,6 +44,13 @@ jobs:
     - name: Checkout code
       uses: actions/checkout@v4
 
+    - name: Get the number of available processes
+      run: |
+        NPROC=$(nproc)
+        echo "NPROC=$NPROC" >> $GITHUB_ENV
+        NPROC_HALF=$((NPROC / 2))
```

Comment:
What happens if NPROC happens to be 1? Probably needs a ternary to make sure NPROC_HALF is not set to 0.

Reply:
@vanderhe oh, and it ran OK for the private repo.

Reply:
I assumed that GitHub never downgrades its runners to a single process, but let me deal with this edge case.

Reply:
@bhourahine The updated version should now correctly handle the `NPROC=1` case.